TargetRenamer/Trados.TargetRenamer/Trados.TargetRenamer/ViewModel/TargetRenamerSettingsViewModel.cs

```diff
@@ -1,9 +1,191 @@
-﻿using Sdl.Core.Settings;
+﻿using System.IO;
+using System.Windows.Input;
+using Sdl.Core.Settings;
+using Sdl.Desktop.IntegrationApi;
+using Trados.TargetRenamer.Interfaces;
+using Trados.TargetRenamer.Model;
+using Trados.TargetRenamer.Services;
 
 namespace Trados.TargetRenamer.ViewModel
 {
-    public class TargetRenamerSettingsViewModel : SettingsGroup
+    public class TargetRenamerSettingsViewModel : BaseModel, ISettingsAware<TargetRenamerSettings>
     {
-	    public TargetRenamerSettings Settings;
-    }
+	    public TargetRenamerSettings Settings { get; set; }
+
+	    private bool _overwriteTargetFiles = true;
+	    private bool _appendAsPrefix;
+	    private bool _appendAsSuffix = true;
+	    private bool _useRegularExpression;
+	    private bool _useCustomLocation;
+	    private string _customLocation = Path.GetTempPath();
+	    private string _regularExpressionSearchFor;
+	    private string _regularExpressionReplaceWith;
+	    private string _delimitator = "_";
+	    private bool _useShortLocales;
+	    private bool _appendTargetLanguage = true;
+	    private bool _appendCustomString;
+	    private string _customString;
+	    private readonly IFolderDialogService _folderDialogService;
+	    private ICommand _selectTargetFolder;
+
+	    public TargetRenamerSettingsViewModel(IFolderDialogService folderDialogService)
+	    {
+		    _folderDialogService = folderDialogService;
+	    }
+
+	    public ICommand SelectTargetFolder =>
+		    _selectTargetFolder ?? (_selectTargetFolder = new CommandHandler(SelectFolder));
+
+	    private void SelectFolder(object obj)
+	    {
+		    CustomLocation = _folderDialogService.ShowDialog();
```

Comment:
What happens with the custom location property if the user clicks "Cancel" or "x" button. Is this scenario handled?

Reply:
The custom location is not changed